### ISSUE TITLE
No hinting for imported library with generated bindings.

### ISSUE THREAD

**Issue by maxim-banaev** [intellij-rust/intellij-rust#6134]
## Environment

* **IntelliJ Rust plugin version:**
* **Rust toolchain version:**
* **IDE name and version:** CL-202.6948.80
* **Operating system:** OS Windows 10(amd64) v10.0

## Problem description

I am using the library `godot-rust` and while everything runs fine, there is no intellisense happening for the (generated) types provided by the package.

This can reproduced by cloning my example repository here:
https://github.com/Sibz/godotrust-bugdemo
This may need LLVM installed in order to compile.

On line 15 I have picked an example method that should hint, but does not.
The description for this struct is here: https://docs.rs/gdnative/0.9.0-preview.0/gdnative/api/rigid_body/struct.RigidBody.html

I have contacted the developer and he is aware of the issue but believes the issue is with the code analyzer - it should be able to pick up the bindings generated but it doesn't.

While this is probably an issue with rust-analyzer, I am not sure if CLion uses that for it's intellisense, hence I am reporting the bug here too.
I have reported this on the rust-analyzer repository
https://github.com/rust-analyzer/rust-analyzer/issues/5992

The code compiles and runs fine.

Submitted to CLion bug tracker: https://youtrack.jetbrains.com/issue/CPP-22114

**Comment by Bromeon** [comment]
godot-rust developer here. I can confirm this still happens with CLion `2021.2.1` on Windows 10.
Rust plugin version `0.4.155.4114-212`

**TLDR:** symbols generated during custom `build.rs` scripts are not indexed.
This means that auto-complete, jump to declaration or syntax highlighting are not operational.

I enabled experimental support for procedural macros (you see that new macro engine and build script evaluation are already active):
![experimental-settings](https://user-images.githubusercontent.com/708488/134772720-049f0560-ae91-423a-b2e5-b74fc1bc53e9.png)

Then I ran _Invalidate Caches & Restart_ and deleted the `target` folder:
![restart](https://user-images.githubusercontent.com/708488/134772731-5316033f-95cf-4584-8dfa-2f424a6634c7.png)

But the problem remains.

Possibly related: **#6908** (tracker), #7365, #6875

**Comment by Undin** [comment]
The root problem is that generated file is huge (more than 2.5 MB), the IDE doesn’t try to analyze it because of performance reasons.
![image](https://user-images.githubusercontent.com/2539310/134781153-47fe08b3-2d6e-44b7-83bc-737a908cc1aa.png)
![image](https://user-images.githubusercontent.com/2539310/134781321-865850c0-8e9a-4405-9c91-1d33c45b8020.png)


It’s a common restriction of IntelliJ platform and not specific to the Rust plugin itself. You can change the threshold with `idea.max.intellisense.filesize=%necessary_threshold_in_kb%` [custom property](https://www.jetbrains.com/help/idea/tuning-the-ide.html#procedure-platform-properties). Note, it may affect editor performance and increase memory consumption. See [common-platform-properties](https://www.jetbrains.com/help/idea/tuning-the-ide.html#common-platform-properties) for more details.

JFYI support for generated files included via `include!(concat!(env!("OUT_DIR"), "/generated.rs"));` is not related to procedural macro support so only `org.rust.cargo.evaluate.build.scripts` [experimental feature](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features) is necessary at this moment to collect info about `OUT_DIR` value. See #1908 for more details

**Comment by Bromeon** [comment]
@Undin great hint, that's indeed the cause!

After adding this file in `C:\Users\<me>\AppData\Roaming\JetBrains\CLion2021.2`:
```properties
idea.max.intellisense.filesize=10240 # 10 MB
```
The symbols are recognized, but performance is indeed not great.

If godot-rust generates multiple files and `include!` them, will that still contribute to a too large file in `build.rs` (where they are included)?
Otherwise we could work with modules...
